### PR TITLE
3x tweaks

### PR DESCRIPTION
**Pull request by manolama**

### CODE REVIEW

File: core/src/main/java/net/opentsdb/query/processor/expressions/ExpressionNumericArrayIterator.java [hunk 1]
```diff
@@ -41,9 +41,33 @@
     BaseExpressionNumericIterator<NumericArrayType>
   implements NumericArrayType {
 
+  private static final int STATIC_ARRAY_LEN = 86_400;
```
Comment (smrutilal2) [inline core/src/main/java/net/opentsdb/query/processor/expressions/ExpressionNumericArrayIterator.java:44]: Why and array length of `86_400 `. Isn't that to much of allocation for small time range queries?

Comment (manolama) [reply core/src/main/java/net/opentsdb/query/processor/expressions/ExpressionNumericArrayIterator.java:44]: Just wanted to capture most cases. I'll add some instrumentation eventually and see what the best size is. 1 day of second data should be adequate though :)

Comment (smrutilal2) [reply core/src/main/java/net/opentsdb/query/processor/expressions/ExpressionNumericArrayIterator.java:44]: And we have ThreadLocal allocation. Which is 8 by default, I guess.

Comment (smrutilal2) [reply core/src/main/java/net/opentsdb/query/processor/expressions/ExpressionNumericArrayIterator.java:44]: I guess, it depends how big is the actual data segment. If the segment is smaller For example 2 hours incase of Aura, then `7200` is probably good enough.

**Review by smrutilal2 — APPROVED** [review]

**Pull request merged**